library/cwm/src/lib/cwm/dynamic_progress_bar.rb

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "abstract_method"
+require "cwm/progress_bar"
+
+module CWM
+  # Widget for a dynamic progress bar, where the label can be set for every step
```

Comment:
I would probably mention here where it can be useful. For me it is something like. `Useful when steps are not known in advance or part of step is dynamically generated.` Do I get it right?

Reply:
Yes, that is.